File: omnes.Web/Modules/ServerTypes/Parametros/TiposCuentasBancariaRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface TiposCuentasBancariaRow {
    IdTipoCuentaBancaria?: number;
    Descripcion?: string;
    Baja?: boolean;
    Observaciones?: string;
}

export abstract class TiposCuentasBancariaRow {
    static readonly idProperty = 'IdTipoCuentaBancaria';
    static readonly nameProperty = 'Descripcion';
    static readonly localTextPrefix = 'Parametros.TiposCuentasBancaria';
    static readonly deletePermission = 'SuperAdmin';
    static readonly insertPermission = 'SuperAdmin';
    static readonly readPermission = 'SuperAdmin';
    static readonly updatePermission = 'SuperAdmin';

    static readonly Fields = fieldsProxy<TiposCuentasBancariaRow>();
}

File: omnes.Web/Modules/Parametros/TiposCuentasBancaria/TiposCuentasBancariaRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace omnes.Parametros;

[ConnectionKey("Default"), Module("Parametros"), TableName("TiposCuentasBancaria")]
[DisplayName("Tipos Cuentas Bancaria"), InstanceName("Tipos Cuentas Bancaria")]
[ReadPermission("SuperAdmin")]
[ModifyPermission("SuperAdmin")]
public sealed partial class TiposCuentasBancariaRow : Row<TiposCuentasBancariaRow.RowFields>, IIdRow, INameRow
{
    class RowTemplate
    {
        [DisplayName("Id Tipo Cuenta Bancaria"), Identity, IdProperty]
        public int? IdTipoCuentaBancaria { get; set; }

        [DisplayName("Descripcion"), Size(100), NotNull, QuickSearch, NameProperty]
        public string Descripcion { get; set; }

        [DisplayName("Baja"), NotNull]
        public bool? Baja { get; set; }

        [DisplayName("Observaciones")]
        public string Observaciones { get; set; }
    }
}